packages/rum/test/userActionCollection.spec.ts

```diff
@@ -92,9 +95,35 @@ describe('startUserActionCollection', () => {
     button.click()
 
     clock.expire()
+  }
+
+  it('cancels pending user action on view loading', () => {
+    const fakeLocation: Partial<Location> = { pathname: '/foo' }
+    const fakeSession = {
+      getId() {
+        return '42'
+      },
+    }
+
+    button.addEventListener(DOM_EVENT.CLICK, () => {
+      clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+      // Since we don't collect dom mutations for this test, manually dispatch one
+      lifeCycle.notify(LifeCycleEventType.DOM_MUTATED)
+    })
+
+    clock.tick(SOME_ARBITRARY_DELAY)
+    button.click()
+    startViewCollection(fakeLocation as Location, lifeCycle, fakeSession as RumSession)
```

Comment:
You are doing a unit test, FMU the main goal of unit test is to test the behavior of a component by looking at its outputs by giving it different inputs.
A test that verify the behavior of multiple components together is an integration test, it can be valuable but a downside is that its behavior is coupled to the behavior of all the tested components.

Here, you are testing `startUserActionCollection`.
By invoking `startViewCollection`, your test is now dependent of this component behavior too.
It seems unlikely to me that `startViewCollection` can be called after that `startUserActionCollection` has started to track some clicks.

IMO, we should stick to unit testing `startUserActionCollection` here.
If you really want to do an integration test, I would be curious why and you should have a more realistic setup.